### PR TITLE
Integration test stabilization + update proto and Dapr runtime to 1.12 RC.

### PR DESCRIPTION
**Pull request by artursouza**
# Description

Update Dapr runtime to 1.12 RC.

## Issue reference

We strive to have all PR being opened based on an issue, where the problem or feature have been discussed prior to implementation.

Please reference the issue this PR will close: endgame

## Checklist

Please make sure you've completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Extended the documentation

### CODE REVIEW

**Review by mukundansundar — DISMISSED** [review]

File: .github/workflows/build.yml
Comment (artursouza) [inline .github/workflows/build.yml:50]: Temporary until the fix is merged in runtime. Don't merge this yet.

File: .github/workflows/build.yml
Comment (artursouza) [inline .github/workflows/build.yml:77]: Temporary until the fix is merged in runtime. Don't merge this yet.

File: sdk-tests/src/test/java/io/dapr/it/tracing/http/Service.java
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/tracing/http/Service.java:25]: This is a must to avoid a race condition between invoking a service and it being available.

File: sdk-tests/src/test/java/io/dapr/it/methodinvoke/grpc/MethodInvokeService.java
Comment (artursouza) [inline sdk-tests/src/test/java/io/dapr/it/methodinvoke/grpc/MethodInvokeService.java:177]: This allows the issue to be reproduced locally, almost 100% of the time.

**Review by mukundansundar — DISMISSED** [review]

**Comment by mukundansundar** [comment]
@artursouza Just one question, as a whole we are waiting for Dapr to be completely ready before invoking any service right?

**Comment by artursouza** [comment]
> @artursouza Just one question, as a whole we are waiting for Dapr to be completely ready before invoking any service right?

Yes but with this change, I am also waiting for the app's health endpoint.

File: .github/workflows/build.yml
Comment (artursouza) [inline .github/workflows/build.yml:50]: This is a fix that allows runtime's grpc app channel to recover.

**Comment by artursouza** [comment]
Force merging this to unblock the release.

**Review by mukundansundar — APPROVED** [review]

**Pull request merged**
